diuf/sudoku/Cell.java

```diff
@@ -92,6 +92,21 @@ public CellSet getVisibleCells() {
     	return Grid.visibleCellsSet[index];
     }
 
+   public boolean canSeeCell(Cell other) {
+	   return Grid.visibleCellsSet[index].contains(other);
+   }
+   
+   public boolean canSeeAnyOfCells(CellSet cellSet) {
+	    CellSet currentSet = new CellSet(Grid.visibleCellsSet[index]);
+		currentSet.retainAll(cellSet);
+		int currentSetSize = currentSet.size();
+		if (currentSetSize > 0)
+			return true;
+		return false;
+   }
+   

```

Comment:
This can be implemented w/o bitset cloning using BitSet.intersects(), which isn't implemented in CellSet but is easy to implement.

Reply:
This is on the to do list then

Reply:
Actually I read for the first time the documentation for java built-in containers after you create this project and I started reading the code to see how it is done. This is new matter to me too, but has something in common with C++ containers which I worked with from years.

Reply:
Done.